ledger/src/sigverify_shreds.rs

```diff
@@ -45,13 +57,27 @@ pub fn verify_shred_cpu(packet: &Packet, slot_leaders: &HashMap<Slot, Pubkey>) -
     let Some(data) = shred::layout::get_signed_data(shred) else {
         return false;
     };
-    signature.verify(pubkey.as_ref(), data.as_ref())
+    match data {
+        SignedData::Chunk(chunk) => signature.verify(pubkey.as_ref(), chunk),
+        SignedData::MerkleRoot(root) => {
+            let key = (signature, *pubkey, root);
+            if cache.read().unwrap().get(&key).is_some() {
+                true
+            } else if key.0.verify(key.1.as_ref(), key.2.as_ref()) {
+                cache.write().unwrap().put(key, ());
+                true
+            } else {
```

Comment:
Was thinking whether a metric on the hits vs misses would be useful. 

Seems like the cache is large enough given ~2000 shreds per slot where we shouldn't be evicting too often.